### PR TITLE
DR-2363 New dataset overview page

### PR DESCRIPTION
**Pull request by tlangs**
Its here! A new dataset details page! This will now be the default page users see when they click on a dataset from the home view. There's a link to the dataset contents, and a new button that takes them back to the details page. 

I did my best to match the styles in the mockup, but its not an exact match, and there's a long tail of work to make the small changes to make it look exactly the same. The functionality, however, is there! 

You can check it out on my personal instance: https://jade-tl.datarepo-dev.broadinstitute.org

image
image

### CODE REVIEW

**Comment by tlangs** [comment]
I still need to fix the cypress tests, which are very much going to break as a result of these changes.

File: deploy_local_changes.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+trap swap_back EXIT
+
+function stderr {
+    >&2 echo "$@"
+}
+
+function help() {
+  stderr "This little script makes it easy to switch the Dockerfile and Dockerfile.direct files "
+  stderr "and deploy local UI changes to a personal deployment. "
+  stderr "The script requires a 'skaffold.yml' file to be present and set up to point to a personal deployment. "
```
Comment (snf2ye) [inline deploy_local_changes.sh:12]: nice!

**Review by snf2ye — APPROVED** [review]
All the code looks reasonable to me and the UI looks great!

Thanks for providing example datasets on your deployment. You probably already thought of these, but a couple of edge cases to test with a bit more data:
- What happens when the schema is long - does the scrolling work as expected?
- Does the new snapshot list page work as expected when there are bunch of snapshots? (paging, sorting, etc.)

**Comment by tlangs** [comment]
If the schema is long enough, then it will simply extend the bottom of the page. The panel itself won't scroll, but the page will extend to fit the contents.

As for snapshots, I've wired in the pagination code. Getting enough snapshots to get the pagination to show up, however, would take a lot of snapshots. It should work the same as every other `JadeTable`, and I was able to re-use the existing `getSnapshots` saga code, so I don't see why it wouldn't work. 

If it does break, I think we could do a fast follow on. The changes in here are already substantial, and I'm anxious to get them in before develop diverges too much.

**Comment by snf2ye** [comment]
@tlangs  - Sounds great - I'm good with proceeding!

File: src/components/ManageUsersModal.jsx [hunk 2]
```diff
@@ -42,6 +43,19 @@ const styles = (theme) => ({
   openButton: {
     width: '100%',
   },
+  iconButton: {
+    border: `1px solid ${theme.palette.primary.main}`,
+    borderRadius: '3px',
+    padding: '1px',
+    backgroundColor: theme.palette.primary.main,
```
Comment (nmalfroy) [inline src/components/ManageUsersModal.jsx:50]: Is this not standard button styling?

Comment (nmalfroy) [reply src/components/ManageUsersModal.jsx:50]: I think if you give it the primary variant, the normal button styling should work

File: src/components/dataset/details/DatasetDetailPanel.jsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+import { renderCloudPlatforms, renderStorageResources } from '../../../libs/render-utils';
+import DatasetAccess from '../DatasetAccess';
+import DatasetSnapshotsTable from '../../table/DatasetSnapshotsTable';
+
+const styles = (theme) => ({
+  root: {
+    flexGrow: 1,
+  },
+  tabsRoot: {
+    color: '#333F52',
```
Comment (nmalfroy) [inline src/components/dataset/details/DatasetDetailPanel.jsx]: This color should likely be coming coming from the theme

File: src/components/ManageUsersModal.jsx (hunk 2 above)
Comment (nmalfroy) [reply src/components/ManageUsersModal.jsx:50]: sorry, `contained` variant

File: src/components/dataset/query/sidebar/panels/InfoViewDatasetAccess.jsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+);
+
+const helpContent = (
+  <Fragment>
+    <Typography variant="h6">Stewards</Typography>
+    <Typography>
+      Creating a dataset makes a user the Steward (or owner) of it. The steward of a dataset can:
+    </Typography>
+    <ul>
+      <li>Define the schema for a dataset</li>
```
Comment (nmalfroy) [inline src/components/dataset/query/sidebar/panels/InfoViewDatasetAccess.jsx]: Hmm...the implies that the dataset's schema can be updated...

File: src/components/table/DatasetSnapshotsTable.jsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  handleFilterSnapshots = (limit, offset, sort, sortDirection) => {
+    const { dispatch, dataset } = this.props;
+    dispatch(getDatasetSnapshots(limit, offset, sort, sortDirection, [dataset.id]));
+  };
+
+  render() {
+    const { classes, snapshotCount, snapshots, summary } = this.props;
+    // TODO add back modified_date column
```
Comment (nmalfroy) [inline src/components/table/DatasetSnapshotsTable.jsx]: Why this TODO?

**Review by nmalfroy — APPROVED** [review]
Looks so cool!  A couple of small comments that would be good to fix before merging.  Also, it's probably worth sitting down with Lou to make sure that this lines up with what he's thinking.  I'm noticing some general UX strangeness (like the fact that there's no side bar on details is odd to me

File: src/components/dataset/query/sidebar/panels/InfoViewDatasetAccess.jsx (hunk 4 above)
Comment (tlangs) [reply src/components/dataset/query/sidebar/panels/InfoViewDatasetAccess.jsx]: This is pulled directly from our documentation. I'll remove it now, but we should update this when we do add schema migration

File: src/components/ManageUsersModal.jsx (hunk 2 above)
Comment (tlangs) [reply src/components/ManageUsersModal.jsx:50]: No, this is a lot of custom styling to force the icon button to be the same size to the left of the header.

**Comment by tlangs** [comment]
@nmalfroy I removed a _bunch_ of unused code. Let me know if you think it should still be there.

**Comment by nmalfroy** [comment]
That's cool man!  As long as we click test I'm aways down with red lines in PRs :)
